File: _drafts/advent_of_code_2022_ruby_crystal_rust/day_1/day_1_1.rb
the_most_calories = File.read('input.txt')
                        .split(/\n\n/m)
                        .map do |group|
  group
    .strip
    .split(/\n/)
    .map(&:to_i)
end
                        .map(&:sum)
                        .max

puts the_most_calories

File: _drafts/advent_of_code_2022_ruby_crystal_rust/day_1/day_1_1.rs
use std::fs;

fn parse_and_search_calories(filename: &str) -> i32 {
    let text = fs::read_to_string(filename).unwrap();

    text.split("\n\n")
        .map(|group| {
            group
                .trim()
                .split("\n")
                .map(|line| line.parse::<i32>().unwrap())
        })
        .map(|numbers| numbers.sum())
        .max().unwrap()
}

fn main() {
    print!("{}\n", parse_and_search_calories("input.txt"));
}
